source/MaterialXGenShader/TypeDesc.cpp

```diff
@@ -86,4 +144,77 @@ TYPEDESC_REGISTER_TYPE(MATERIAL, "material")
 
 } // namespace Type
 
+
+
+///////////////////////////////////////////////////////////////////
```

Comment:
As above, I would omit these triple lines of forward slashes, and you might instead add a comment noting that these are "StructTypeDesc methods", as seen in examples like the following:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXCore/Element.cpp#L52